mailscanner/functions.php

```diff
@@ -2664,11 +2656,17 @@ function ldap_authenticate($user, $password)
                     return null;
                 }
 
-                if (isset($result[0]['userprincipalname'][0])) {
-                    $user = $result[0]['userprincipalname'][0];
-                } else {
+                if (!isset($result[0][LDAP_USERNAME_FIELD]) ||
+                      !isset($result[0][LDAP_USERNAME_FIELD][0])) {
+                    @trigger_error(__('ldapno03') . ' "' . LDAP_USERNAME_FIELD . '" ' . __('ldapresults03'));
+                    return null;
+                }
+
+                if (LDAP_USERNAME_FIELD === 'cn') {
                     // build DN for user, when LDAP server is not an AD
                     $user = 'cn=' . $result[0]['cn'][0] . ',' . LDAP_DN;
+                } else {
+                    $user = $result[0][LDAP_USERNAME_FIELD][0];
```

Comment:
This way, the $result will just be the field filled in, but a search needs the
complete DN.

for me this is needed;

> $user = LDAP_USERNAME_FIELD . '=' . $result[0][LDAP_USERNAME_FIELD][0] . ',' . LDAP_DN;